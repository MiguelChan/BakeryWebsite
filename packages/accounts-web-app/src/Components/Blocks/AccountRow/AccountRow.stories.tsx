import React from 'react';
import {
  Story,
  Meta,
} from '@storybook/react';
import { Table } from '@mui/material';
import {
  Account,
  AccountType,
} from '@mgl/shared-components';
import {
  AccountRow,
  AccountRowProps,
} from './AccountRow';

export default {
  title: 'Components/Blocks/AccountRow',
  component: AccountRow,
  decorators: [
    (Story) => (
      <Table>
        <tbody>
          <Story />
        </tbody>
      </Table>
    ),
  ],
} as Meta;

const Template: Story<AccountRowProps> = (args) => <AccountRow {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  account: {
    accountType: AccountType.Expenses,
    id: 'ThisIsAnId',
    subAccounts: [],
    title: 'Cuenta de gastos',
  } as Account,
};
