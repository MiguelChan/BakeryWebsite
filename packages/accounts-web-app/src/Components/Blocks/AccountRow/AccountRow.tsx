import React from 'react';
import {
  Account,
} from '@mgl/shared-components';
import {
  TableCell,
  TableRow,
} from '@mui/material';

export type OnAccountClickedListener = (account: Account) => void;

export interface AccountRowProps {
  account: Account;
  onAccountClickedListener?: OnAccountClickedListener;
}

/**
 * Defines a row that is meant to be used to displaying an {Account}.
 *
 * @param {Account} account The account to use.
 *
 * @returns
 */
export const AccountRow: React.FunctionComponent<AccountRowProps> = ({
  account,
  onAccountClickedListener,
}) => {
  const {
    accountType,
    id,
    title,
  } = account;

  const handleClick = (): void => {
    if (onAccountClickedListener) {
      onAccountClickedListener(account);
    }
  };

  return (
    <TableRow
      onClick={handleClick}
      hover
    >
      <TableCell>
        {id}
      </TableCell>
      <TableCell>
        {title}
      </TableCell>
      <TableCell>
        {accountType}
      </TableCell>
    </TableRow>
  );
};
