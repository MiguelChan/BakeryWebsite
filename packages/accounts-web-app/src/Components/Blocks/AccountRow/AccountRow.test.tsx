import React from 'react';
import {
  RenderResult,
  render,
} from '@testing-library/react';
import {
  Account,
  AccountType,
} from '@mgl/shared-components';
import { Table } from '@mui/material';
import { AccountRow } from './AccountRow';

describe('AccountRow', () => {
  function setupComponent(account: Account): RenderResult {
    return render(
      <Table>
        <tbody>
          <AccountRow account={account} />
        </tbody>
      </Table>,
    );
  }

  it('Should display the AccountInformation', () => {
    const fakeAccount: Account = {
      accountType: AccountType.Expenses,
      id: 'ThisIsAnId',
      subAccounts: [],
      title: 'This is an account',
    };

    const result = setupComponent(fakeAccount);

    expect(result.container).toMatchSnapshot();
  });
});
