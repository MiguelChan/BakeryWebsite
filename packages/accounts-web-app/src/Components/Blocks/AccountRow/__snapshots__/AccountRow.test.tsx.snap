// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountRow Should display the AccountInformation 1`] = `
<div>
  <table
    class="MuiTable-root css-rqglhn-MuiTable-root"
  >
    <tbody>
      <tr
        class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
        >
          ThisIsAnId
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
        >
          This is an account
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
        >
          Expenses
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
